README.md

```diff
@@ -68,3 +68,57 @@ The complete incident body can be shown without ever leaving the dashboard:
 
 Besides Grafana, the plugin just needs a running Bosun instance. Because Bosun doesn't have support for [CORS headers](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing), it may be easier to make it work in proxy mode.
 Bosun also needs a ElasticSearch backend in order for its annotations subsystem to work.
+
+## Query Templating
+
+For users unfamiliar with Bosun syntax, or for rapid prototyping, query template blocks can be used.
+
+Simply click the Query + button and a new query block will appear.
+
+![Query template example](src/img/basic_templated_query.png)
+
+In the above example the query
+
+`
+q("zimsum:$ds-:this.is.a.metric{country=UK}{}", "$start", "")
+`
+
+is templated. The variable name is given as $q - variable names must be prepended with a dollar. This can then be called in the Final Query section.
+
+If we wished to also make the value for the country tag a templated variable then we could click the Var + button. This adds a new simple variable block at the bottom. Blocks are read and their variables substituted in order, so to use this new $country block we can click and drag it above the query block.
+This is shown in the gif below, beginning from the query shown above. The resulting query will be functionally equivalent.
+
+![Click and Drag Ex](src/img/reordering_variable_demo.gif)
+
+The free text box still works as before and no templating is required, templated variables and queries are just substituted in when referenced with their name - where the name begins with $.
+
+You can copy the substituted query straight to your clipboard with the 'copy substituted query' button - useful for building a Bosun query string without worrying about the syntax.
+ 
+## Local Development
+
+Clone Grafana. 
+
+Clone this repo into Grafana's plugins folder.
+
+Run `npm install -g grunt-cli`
+
+Add the following to datasources.yml in Grafana to provide the plugin with a suitable datasource:
+```
+  - name: Bosun
+    type: bosun-datasource
+    access: proxy
+    url: <Bosun URL>
+    jsonData:
+      openTSDBUrl: <OpenTSDB URL>
+```
+Autocompletion suggestions for metrics are taken from the openTSDBUrl parameter provided. To prevent CORS errors, run the following:
```

Comment:
I'm being really pedantic but something with this sentence grates with me a little.  Rather than saying:

> are taken from the openTSDBUrl parameter provided

would it be clearer to say something like:

> are taken from the OpenTSDB instance addressed by the provided openTSDBUrl parameter

?